src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -116,19 +115,48 @@ namespace {
          {"SGAS" , {"BSGAS"}}
     };
 
+    using keyword_set = std::unordered_set<std::string>;
+
+    inline bool is_in_set(const keyword_set& set, const std::string& keyword) {
+        return set.find(keyword) != set.end();
+    }
+
     bool is_special(const std::string& keyword) {
-        using set = std::unordered_set<std::string>;
-        static const auto specialkw = set {
-            "NEWTON", "NAIMFRAC", "NLINEARS", "NLINSMIN", "NLINSMAX",
-            "ELAPSED", "MAXDPR", "MAXDSO", "MAXDSG", "MAXDSW", "STEPTYPE",
+        static const keyword_set specialkw {
+            "ELAPSED",
+            "MAXDPR",
+            "MAXDSG",
+            "MAXDSO",
+            "MAXDSW",
+            "NAIMFRAC",
+            "NEWTON",
+            "NLINEARS",
+            "NLINSMAX",
+            "NLINSMIN",
+            "STEPTYPE",
             "WNEWTON",
         };
 
-        return specialkw.count(keyword) > set::size_type{0};
+        return is_in_set(specialkw, keyword);
     }
 
     bool is_udq_blacklist(const std::string& keyword) {
-        return (keyword == "SUMTHIN") || (keyword == "RUNSUM");
+        static const keyword_set udq_blacklistkw {
+            "SUMTHIN",
+        };
+
+        return is_in_set(udq_blacklistkw, keyword);
+    }
+
+    bool is_processing_instruction(const std::string& keyword) {
+        static const keyword_set processing_instructionkw {
+            "NARROW",
+            "RUNSUM",
+            "SEPARATE",
+            "SUMMARY",
+        };
```

Comment:
The concept of a `processing_instructionkw` is good - you can add `RPTONLY` to the list.

Reply:
Fixed.